src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test/Formatting/HtmlFormattingTest.cs

```diff
@@ -983,6 +982,73 @@ @typeparam TValue
             }
         }
 
+        [Fact]
+        [WorkItem("https://github.com/dotnet/razor-tooling/issues/6110")]
+        public async Task FormatExplicitCSharpInsideHtml()
+        {
+            await RunFormattingTestAsync(
+input: @"
+@using System.Text;
+
+<div>
+    @(new C()
```

Comment:
Can we add an example which utilizes Razor templates (`@<tag>..newlines..</tag>`) in them (passing them as args & capturing as locals)? Historically they've been used with explicit expressions to make things slightly more readable but ultimately lead to a bit of insanity in the formatting realm.

Wouldn't be surprised if WebTools formatting tests have a few examples of this as well

Reply:
I might be doing something wrong here, but I'm trying to test/break this, and can't.

Given:
<img width="181" alt="image" src="https://user-images.githubusercontent.com/754264/163086960-f6c54633-4c63-4a08-bf8f-905584008cdd.png">

1. not an explicit statement, so formats without this PR (and seems okay)
2. not valid
3. not valid

Reply:
This was the only interesting example I found in the webtools tests, so I added it to my test.

```
@(   Html.DisplayNameFor (@<text>
        <p >
            <h2 ></h2>
        </p>
        </text>)
        .ToString())
```

Reply:
In that case imma merge